commit 943a5a7283da44ed1135412dd0f2dcd260ddabce (refs/changes/47/2347/1)
Author: Trent Mick <trentm@gmail.com>
Date:   2017-08-08T15:29:11-07:00 (2 years, 2 months ago)
    
    joyent/node-triton#201 Assuming an RBAC role in triton CLI v5 throws error

diff --git a/CHANGES.md b/CHANGES.md
index e495370..4d36359 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -7,9 +7,13 @@ Known issues:
 
 ## not yet released
 
+- [joyent/node-triton#201] Fix `triton -r,--role ROLE ...` option for taking up
+  an RBAC role. This was introduced in v4.12.0 and was accidentally broken
+  in v5.0.0.
 - [joyent/node-triton#217] `triton volume ls -l` should output a `RESOURCE`
   column.
 
+
 ## 5.3.0
 
 - [joyent/node-triton#173], [joyent/node-triton#174] and
diff --git a/lib/config.js b/lib/config.js
index f1cae3f..fe2fda0 100644
--- a/lib/config.js
+++ b/lib/config.js
@@ -62,6 +62,7 @@ var PROFILE_FIELDS = {
     keyId: true,
     insecure: true,
     user: true,
+    roles: true,
     actAsAccount: true
 };
 
@@ -234,6 +235,7 @@ function validateProfile(profile, profilePath) {
             profile.name === 'env' ? 'TRITON_USER or SDC_USER'
                 : 'profile.user');
         assert.optionalString(profile.actAsAccount, 'profile.actAsAccount');
+        assert.optionalArrayOfString(profile.roles, 'profile.roles');
     } catch (err) {
         var msg = format('invalid %sprofile%s: %s',
             profile.name ? '"' + profile.name + '" ' : '',
